capa-prepare-cloud-provider-accounts (#1672)
giantswarm/docs: src/content/getting-started/cloud-provider-accounts/cluster-api/aws/index.md

aliases:
  - /guides/prepare-aws-account-for-tenant-clusters/
  - /guides/prepare-aws-account/
owner:
  - https://github.com/orgs/giantswarm/teams/team-hydra
last_review_date: 2023-01-10
---

The Cluster API workload clusters (the clusters running your Kubernetes workloads) in a Giant Swarm installation can run in an AWS account separate from the management cluster. This gives great flexibility depending on requirements and usage scenarios. For example, it allows the management cluster to be running in an AWS account dedicated to it, whilst workload clusters operate in several different AWS accounts, depending
on the customer's needs.

## Overview

In order to run Giant Swarm Cluster API clusters, all AWS account(s) need to fulfil these requirements (there is [repository](https://github.com/giantswarm/giantswarm-aws-account-prerequisites) that can help with creating the necessary resources in AWS)

- Create [GiantSwarmAdmin](https://github.com/giantswarm/giantswarm-aws-account-prerequisites/tree/master/admin-role) IAM role, which is used by Giant Swarm employees to support and debug Cluster API workload clusters via AWS console.

> - Create [capa-controller-role](https://github.com/giantswarm/giantswarm-aws-account-prerequisites/tree/master/capa-controller-role) IAM role, which is used the controllers and operators to create necessary AWS resources for the Cluster API workload clusters.

Reviewer: ```suggestion
- Create [capa-controller-role](https://github.com/giantswarm/giantswarm-aws-account-prerequisites/tree/master/capa-controller-role) IAM role, which is used by the controllers and operators to create necessary AWS resources for the Cluster API workload clusters.
```

---
linkTitle: AWS
title: Prepare an AWS account to run Cluster API Giant Swarm clusters
description: This guide will walk you through all necessary steps to set up an Amazon AWS account with appropriate IAM roles for operating Cluster API Giant Swarm clusters.
weight: 10
menu:
  main:
    identifier: gettingstarted-cloudprovider-clusterapi-aws
    parent: gettingstarted-cloudprovider-clusterapi
user_questions:
  - What are the recommended service limit/quotas for AWS accounts?
  - How prepare AWS account for the Cluster API clusters?
aliases:
  - /guides/prepare-aws-account-for-tenant-clusters/
  - /guides/prepare-aws-account/
owner:
  - https://github.com/orgs/giantswarm/teams/team-hydra
last_review_date: 2023-01-10
---

> The Cluster API workload clusters (the clusters running your Kubernetes workloads) in a Giant
> Swarm installation can run in an AWS account separate from the management cluster.
> This gives great flexibility depending on requirements and usage
> scenarios. For example, it allows the management cluster to be running in an AWS account
> dedicated to it, whilst workload clusters operate in several different AWS accounts, depending
> on the customer needs.

Reviewer 2: ```suggestion
The Cluster API workload clusters (the clusters running your Kubernetes workloads) in a Giant Swarm installation can run in an AWS account separate from the management cluster. This gives great flexibility depending on requirements and usage scenarios. For example, it allows the management cluster to be running in an AWS account dedicated to it, whilst workload clusters operate in several different AWS accounts, depending
on the customer's needs.
```
Spacing was weird

## Overview

> In order to run Giant Swarm Cluster API clusters, the all AWS account(s) need to fulfill
> these requirements (there is [repository](https://github.com/giantswarm/giantswarm-aws-account-prerequisites) that can help with creating the necessery resources in AWS)
> - create [GiantSwarmAdmin](https://github.com/giantswarm/giantswarm-aws-account-prerequisites/tree/master/admin-role) IAM role, which is used by Giant Swarm employees to support and debug Cluster API workload clusters via AWS console.
> - create [capa-controller-role](https://github.com/giantswarm/giantswarm-aws-account-prerequisites/tree/master/capa-controller-role) IAM role, which is used the controllers and operators to create necessery AWS resources for the Cluster API workload clusters.
> - Service limits set according to requirements.
> 
> We have created a Terraform module to automate the IAM role creation. You can view the code [here](https://github.com/giantswarm/giantswarm-aws-account-prerequisites). You can also use the steps as described in this guide.

Reviewer 2: ```suggestion
In order to run Giant Swarm Cluster API clusters, all AWS account(s) need to fulfil these requirements (there is [repository](https://github.com/giantswarm/giantswarm-aws-account-prerequisites) that can help with creating the necessary resources in AWS)

- Create [GiantSwarmAdmin](https://github.com/giantswarm/giantswarm-aws-account-prerequisites/tree/master/admin-role) IAM role, which is used by Giant Swarm employees to support and debug Cluster API workload clusters via AWS console.
- Create [capa-controller-role](https://github.com/giantswarm/giantswarm-aws-account-prerequisites/tree/master/capa-controller-role) IAM role, which is used the controllers and operators to create necessary AWS resources for the Cluster API workload clusters.
- Service limits are set according to requirements.

We have created a Terraform module to automate the IAM role creation. You can view the code [here](https://github.com/giantswarm/giantswarm-aws-account-prerequisites). You can also use the steps as described in this guide.
```

## Service limits in AWS accounts {#limits}

> A number of limits apply to an AWS account initially, which are described in the
> [AWS Service Limits documentation](https://docs.aws.amazon.com/general/latest/gr/aws_service_limits.html).
> The following overview lists the limits that have to be adjusted in order to use
> the account to operate Giant Swarm Cluster API workload clusters.
> 
> Adjusting a service limit requires a support case in the
> [AWS Support Center](https://console.aws.amazon.com/support/home),
> where a specific entry form is provided for this type of case. Each limit type
> requires a separate case. When creating these, make sure to be logged in to the
> AWS account you want to adjust the limits for, and always select the correct
> region.

Reviewer 2: ```suggestion
A number of limits apply to an AWS account initially, which are described in the [AWS Service Limits documentation](https://docs.aws.amazon.com/general/latest/gr/aws_service_limits.html). The following overview lists the limits that have to be adjusted in order to use the account to operate Giant Swarm Cluster API workload clusters.

Adjusting a service limit requires a support case in the [AWS Support Center](https://console.aws.amazon.com/support/home), where a specific entry form is provided for this type of case. Each limit type requires a separate case. When creating these, make sure to be logged in to the AWS account you want to adjust the limits for, and always select the correct region.
```

    - Elastic IP
        - New VPC Elastic IP Address Limit per region: **50**

- AWS account intended to run private clusters (outgoing traffic goes to public internet only thru proxy)
    - Route 53 Resolver
        - Endpoints per AWS Region: **100**

(Please extend the list of EC2 instances to also contain the types you need frequently.)

## Configure the `AWSClusterRoleIdentity`
When you want to create a new Cluster API workload cluster in a new AWS account you need to create a CR `AWSClusterRoleIdentity` which will reference the role in the new AWS Account.

```yaml
apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
kind: AWSClusterRoleIdentity
metadata:
  labels:
    cluster.x-k8s.io/watch-filter: capi
  name: ${ROLE_NAME}
spec:
  allowedNamespaces:
    list: null
    selector: {}
  roleARN: ${ROLE_ARN}
  sourceIdentityRef:
    kind: AWSClusterControllerIdentity
    name: default
```
Where:
* `${ROLE_NAME}` is a short unique name referencing AWS account (ie: `development`, `sandbox` or `staging2`)
* `${ROLE_ARN}` is full ARN of the role created in previous step with `giantswarm-aws-account-prerequisites` repository.

> This CR needs to be created only once for each AWS Account. It can be then refered in the `cluster-aws` repo in the value [`aws.awsClusterRoleIdentityName`](https://github.com/giantswarm/cluster-aws/blob/master/helm/cluster-aws/values.yaml#L14).

Reviewer 2: ```suggestion
This CR needs to be created only once for each AWS Account. It can be then referenced in the `cluster-aws` repo in the value [`aws.awsClusterRoleIdentityName`](https://github.com/giantswarm/cluster-aws/blob/master/helm/cluster-aws/values.yaml#L14).
```
